src/core/src/main/java/org/apache/jmeter/threads/TestCompiler.java

```diff
@@ -203,7 +203,8 @@ private void saveSamplerConfigs(Sampler sam) {
         List<Assertion> assertions = new ArrayList<>();
         List<PostProcessor> posts = new ArrayList<>();
         List<PreProcessor> pres = new ArrayList<>();
-        for (int i = stack.size(); i > 0; i--) {
+        int stackSize = stack.size();
+        for (int i = stackSize; i > 0; i--) {
```

Comment:
This change is not justified. `int i = stack.size();` was executed only once when the loop was initialized, so adding extra variable changes nothing here.

Reply:
Fixed